homeassistant/components/weather/__init__.py

```diff
@@ -114,52 +190,125 @@ class WeatherEntity(Entity):
     _attr_humidity: float | None = None
     _attr_ozone: float | None = None
     _attr_precision: float
-    _attr_pressure: float | None = None
+    _attr_native_pressure: float | None = None
     _attr_pressure_unit: str | None = None
+    _attr_native_pressure_unit: str | None = None
     _attr_state: None = None
-    _attr_temperature_unit: str
-    _attr_temperature: float | None
-    _attr_visibility: float | None = None
+    _attr_temperature_unit: str | None = None
+    _attr_native_temperature_unit: str
+    _attr_native_temperature: float
+    _attr_native_visibility: float | None = None
     _attr_visibility_unit: str | None = None
+    _attr_native_visibility_unit: str | None = None
+    _attr_native_precipitation: float | None = None
     _attr_precipitation_unit: str | None = None
+    _attr_native_precipitation_unit: str | None = None
     _attr_wind_bearing: float | str | None = None
-    _attr_wind_speed: float | None = None
+    _attr_native_wind_speed: float | None = None
     _attr_wind_speed_unit: str | None = None
+    _attr_native_wind_speed_unit: str | None = None
```

Comment:
Let's keep these sorted for readability, maybe with _attr_native_* in it's own section.